src/clientAuth/bearerHeader.js

```diff
@@ -6,49 +12,47 @@ class BearerHeader extends ClientAuthnMethod {
   }
 
   /**
-   * More complicated logic then I would have liked it to be
-   * @param {*} cis Request class instance
-   * @param {*} ci Client information
-   * @param {*} requestArgs: request arguments
-   * @param {*} httpArgs HTTP header arguments
-   * @param {*} kwargs
+   * @param {?ResourceRequest} cis Request class instance
+   * @param {?ClientInfo} ci Client information
+   * @param {?Object.<string, string>} requestArgs Request arguments
+   * @param {?Object.<string, string>} httpArgs HTTP header arguments
+   * @param {?Object.<string, string>} kwargs Other optional arguments
    */
   construct(cis, cliInfo, httpArgs, kwargs) {
     cis = cis || null;
     cliInfo = cliInfo || null;
     httpArgs = httpArgs || null;
 
-    let origCis = cis;
+    const origCis = cis;
     let accToken = null;
     if (cis !== null) {
-      if (Object.keys(cis).indexOf('access_token') !== -1) {
-        accToken = cis['access_token'];
-        delete cis['access_token'];
+      if (cis.access_token) {
+        accToken = cis.access_token;
+        delete cis.access_token;
```

Comment:
Deleting the property from `cis` will also delete it from `origCis`, why creating this `origCis` variable?